compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -239,11 +257,75 @@ LogicalResult AIETargetBackend::serializeExecutable(
     for (auto s : cmdArgs) llvm::dbgs() << s << " ";
     llvm::dbgs() << "\n";
   }
-  int result = llvm::sys::ExecuteAndWait(cmdArgs[0], cmdArgs, cmdEnv);
-  if (result != 0) {
-    return moduleOp.emitOpError(
-        "Failed to produce an XCLBin with external tool");
+
+  auto getTempFileName =
```

Comment:
This is great to handle and output/error messaging here will definitely help our development velocity. I have following questions 
1. why cant the tool do this on its side and we need to do it here?
2. If we must do it here, can we somehow make it less visible by putting it in a different function, as when reading the code and understanding the flow, a reader might not necessarily care about these low level IO)handling details.

Reply:
Thanks Nirvedh, 

1. Which tool, do you mean aie2xclbin? I added the ability for aie2xclbin to print in https://github.com/Xilinx/mlir-aie/pull/1058, this current PR just propagates those options to iree-compile. 

2. Agree, I'll try and refactor.

Reply:
(a test would definitely make point 1 clearer)

Reply:
About 1. My question is about handling the stdout/stderr here, I was thinking we pass the flags from here and the tool should by default give the output/error streams ?

Reply:
Ok I think I understand your question now. 

We call aie2xclbin through llvm::sys::ExecuteAndWait, which doesn't seem to have the option to pipe the stdout/stderr from the called program. It only has the ability to write them to files which is what this new code does:

```
  std::array<std::optional<StringRef>, 3> redirects{
      /* std::cin = */
      std::optional<StringRef>{}, stdoutFile.str(), stderrFile.str()};

  int result =
      llvm::sys::ExecuteAndWait(cmdArgs[0], cmdArgs, cmdEnv, redirects);
 ```
 
 
 I looked around for an alternative which doesn't involve temporary files, but couldn't find any. The approach I've implemented here is similar to that in IREE here: https://github.com/openxla/iree/blob/e6397cb853de7f3b28dcdc05987c40f0a4d1e9fe/compiler/plugins/target/CUDA/CUDATarget.cpp#L129

Reply:
Cool, thanks for explaining!